Fusion type checking (#8785)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/run_query.md

**Note:** The `run_query` macro will not begin a transaction automatically - if you wish to run your query inside of a transaction, please use `begin` and `commit ` statements as appropriate.

<VersionBlock firstVersion="2.0">

## Fusion type checking

> The <Constant name="fusion_engine" /> processes result sets with strict null checking. This can cause failures when using `run_query` for DDL or utility operations (like `OPTIMIZE`, `VACUUM`, or maintenance queries) that return result sets with null values in columns declared as non-nullable.

Reviewer: ```suggestion
The <Constant name="fusion_engine" /> processes result sets with strict null checking. This can cause failures when using `run_query` for <Term id="ddl" /> or utility operations (like `OPTIMIZE`, `VACUUM`, or maintenance queries) that return result sets with null values in columns declared as non-nullable.
```

Returns a [Table](https://agate.readthedocs.io/page/api/table.html) object with the result of the query. If the specified query does not return results (for example, a <Term id="ddl" />, <Term id="dml" />, or maintenance query), then the return value will be `none`.

<VersionBlock firstVersion="2.0">

## Fusion type checking

The <Constant name="fusion_engine" /> processes result sets with more strict null checking. This can cause failures when using `run_query` for <Term id="ddl" /> or maintenance operations (like `OPTIMIZE`, `VACUUM`). When a given result set returns null values in columns declared as non-nullable, fusion will fail - whereas dbt Core would silently ignore it.

> For "fire and forget" operations where you don't need the result set, use a [statement block](/reference/dbt-jinja-functions/statement-blocks) with `fetch_result=False` instead:

Author: ```suggestion
For "fire and forget" operations where you don't need the result set, use a [statement block](/reference/dbt-jinja-functions/statement-blocks) with `fetch_result=False` instead, as highlighted in this Databricks example:
```

Returns a [Table](https://agate.readthedocs.io/page/api/table.html) object with the result of the query. If the specified query does not return results (for example, a <Term id="ddl" />, <Term id="dml" />, or maintenance query), then the return value will be `none`.

<VersionBlock firstVersion="2.0">

## Fusion type checking

> The <Constant name="fusion_engine" /> processes result sets with strict null checking. This can cause failures when using `run_query` for <Term id="ddl" /> or utility operations (like `OPTIMIZE`, `VACUUM`, or maintenance queries) that return result sets with null values in columns declared as non-nullable.

Reviewer 2: ```suggestion
The <Constant name="fusion_engine" /> processes result sets with more strict null checking. This can cause failures when using `run_query` for <Term id="ddl" /> or maintenance operations (like `OPTIMIZE`, `VACUUM`). When a given result set returns null values in columns declared as non-nullable, fusion will fail - whereas dbt Core would silently ignore it.
```

For "fire and forget" operations where you don't need the result set, use a [statement block](/reference/dbt-jinja-functions/statement-blocks) with `fetch_result=False` instead:

> ```jinja

Reviewer 2: Can we highlight this is an example operation for Databricks - I don't want people to copy paste it when they are on another engine, where this query would not run properly

Author: Done!

{% macro run_optimize(table, zorder_fields) %}
  {% set zorder_str = zorder_fields | join(', ') %}

  {% set query %}
    OPTIMIZE {{ table }}
    {% if zorder_str | length > 0 %}
      ZORDER BY ({{ zorder_str }})
    {% endif %}
  {% endset %}

  {% call statement('optimize', fetch_result=False) %}
    {{ query }}
  {% endcall %}
{% endmacro %}
```

> This pattern is particularly relevant for Databricks users running `OPTIMIZE` or `VACUUM` in post-hooks.

Reviewer 2: I would remove this line, just because it's not particular to Databricks what we are suggesting here.
```suggestion
```
